core/src/main/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectLifecycle.scala

```diff
@@ -62,6 +63,10 @@ class GoogleBillingProjectLifecycle(
                                  ctx: RawlsRequestContext
   ): Future[CreationStatus] =
     for {
+      - <- createBillingProfile(createProjectRequest, ctx).flatMap { profileModel =>
+        addMembersToBillingProfile(profileModel, createProjectRequest, ctx)
+        billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
+      }
```

Comment:
These should be individual lines in the for-comp
```suggestion
      profileModel <- createBillingProfile(createProjectRequest, ctx)
      _ <- addMembersToBillingProfile(profileModel, createProjectRequest, ctx)
      _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
```

Reply:
I will make the change as I definitely agree it is easier to read, but functionally is there a difference?

Reply:
There is. Each `<-` in a for-comp is equivalent to a `flatMap`, so you would need an additional `flatMap` call on `addMembersToBillingProfile` for it to be functionally equivalent to my suggestion. 
```
- <- createBillingProfile(createProjectRequest, ctx).flatMap { profileModel =>
        addMembersToBillingProfile(profileModel, createProjectRequest, ctx).flatMap { _ =>
          billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
        }
      }
```

As you had it, Rawls won't wait for `addMembersToBillingProfile` to complete before it moves on to `billingRepository.setBillingProfileId` -- it's essentially dropping the `addMembersToBillingProfile` `Future` on the floor instead of adding it to the chain of `Future`s that will run in order.

You can see it in action using the scala interpreter with something like this

```
scala> import scala.concurrent.Future
scala> implicit val ec: scala.concurrent.ExecutionContext = scala.concurrent.ExecutionContext.global

scala> def slow(): Future[Unit] = Future {
     |   Thread.sleep(500)
     |   println("i'm slow")
     | }
scala> def fast(): Future[Unit] = Future {
     |   Thread.sleep(100)
     |   println("i'm fast")
     | }

scala> for {
     |   _ <- fast().flatMap { _ =>
     |     slow()
     |     fast()
     |   }
     |   _ <- fast()
     | } yield ()

output
i'm fast
i'm fast
i'm fast
i'm slow

scala> for {
     |   _ <- fast()
     |   _ <- slow()
     |   _ <- fast()
     |   _ <- fast()
     | } yield ()

output
i'm fast
i'm slow
i'm fast
i'm fast

Reply:
Ah, thank you for the detailed response! It really helped.